File: OSEUK/algorithm/week2/PGS_H_index.java
package programmers.kit.sort;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class HIndex {
    public static int solution(int[] citations){
        Arrays.sort(citations);
        int hIndex = 0;
        for (int i = citations.length-1; i>=0; i--){
            int h = Math.min(citations[i], citations.length-i);
            hIndex  = Math.max(hIndex, h);
        }

        return hIndex;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] strs = br.readLine().split(" ");
        int[] citations = new int[strs.length];

        for (int i = 0; i < citations.length; i++){
            citations[i] = Integer.parseInt(strs[i]);
        }

        System.out.println(solution(citations));
    }
}

File: cjh-19/algorithm/week7/PGS_H-Index.js
function solution(citations) {
    var answer = 0;
    citations.sort((a,b) => b - a);
    
    for(let i=0; i<citations.length; i++){
        let Hnum = citations[i]; // 현재 논문 인용 횟수
        let top = i+1; // Hnum이상 인용한 논문 수
        if(top <= Hnum){ // 논문수 <= 인용 횟수
            answer = top; // 논문 수로 H-Index 갱신
        }
        else {
            break;
        }
    }
    
    return answer;
}